src/Common/tests/System/PlatformDetection.cs

```diff
@@ -329,6 +331,30 @@ private static int GetWindowsBuildNumber()
             return -1;
         }
 
+        const int PRODUCT_IOTUAP = 0x0000007B;
+        const int PRODUCT_IOTUAPCOMMERCIAL = 0x00000083;
+
+        [DllImport("kernel32.dll", SetLastError = false)]
+        static extern bool GetProductInfo(
+         int dwOSMajorVersion,
+         int dwOSMinorVersion,
+         int dwSpMajorVersion,
+         int dwSpMinorVersion,
+         out int pdwReturnedProductType
+        );
+
+        public static bool IsWindowsIoTCoreEdition()
```

Comment:
I would either make this private, since you have `IsWindowsIoTCore` property as well, or inline this into `IsWindowsIoTCore`.